packages/concerto-core/types/lib/decoratormanager.d.ts

```diff
@@ -37,8 +37,42 @@ declare class DecoratorManager {
      */
     private static canMigrate;
     /**
-     * Applies all the decorator commands from the DecoratorCommandSet
-     * to the ModelManager.
+     * Add decorator commands set with index object to the coresponding target map
+     * @param {*} targetMap the target map to add the command to
+     * @param {targetKey} targetKey the target key to add the command to
+     * @param {DcsIndexWrapper} dcsWithIndex the command to add
+     * @private
+     */
+    private static addDcsWithIndexToMap;
+    /**
+     * Creates five different maps to index decorator command sets by target type and returns them
+     * @param {*} decoratorCommandSet the DecoratorCommandSet object
+     * @returns {Object} a new model manager with the decorations applied
+     * @private
+     */
+    private static getDecoratorMaps;
+    /**
+     * Migrate or validate the DecoratorCommandSet object if the options are set as true
```

Comment:
Migrate AND validate?

Reply:
It can also do either of the operation if parameters for the arguments are passed so. It can only validate or it might only migrate and can do both if the parameters are true for both.

The only reason I moved this out of `decorateModels` is to reduce the code in the `decorateModels` method, so is a private method just used in `decorateModels` method.